File: D1-Morning/C#/D1-S2/W14-OneToMany/W14-OneToMany/Department.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace W14_OneToMany
{
    public class Department
    {
        public string Code { set; get; }
        public string Name { set; get; }

        public List<Student> StudentList = new List<Student>();
        public Department()
        { 

        }
        public Department(string code, string name)
        {
            Code = code;
            Name = name;
        }
    }
}

File: D1-Morning/D1-S2-PHP/P7-LogicImplementation/Department.php
<?php


class Department
{
    private $code;
    private $name;
    private $studentList;

    public function getCode()
    {
        return $this->code;
    }

    public function setCode($code): void
    {
        $this->code = $code;
    }

    public function getName()
    {
        return $this->name;
    }

    public function setName($name): void
    {
        $this->name = $name;
    }

    public function getStudentList()
    {
        return $this->studentList;
    }

    public function setStudentList($studentList): void
    {
        $this->studentList = $studentList;
    }

    public function addStudent($student){

      foreach ($this->studentList as $studentData){
        if($studentData->getRegNo() == $student->getRegNo()){
          return 'Duplicate entry';
        }
      }

      array_push($this->studentList,$student);

      return "Added";

    }


    public function __construct($code, $name)
    {
        $this->code = $code;
        $this->name = $name;
    }

}